infra/Pos/Network/Yaml.hs

```diff
@@ -251,3 +273,157 @@ instance FromJSON Topology where
 
 instance IsConfig Topology where
   configPrefix = return Nothing
+
+{-------------------------------------------------------------------------------
+  Policies described in JSON/YAML.
+-------------------------------------------------------------------------------}
+
+-- | Policies described by a JSON/YAML.
+data StaticPolicies = StaticPolicies {
+      staticEnqueuePolicy :: StaticEnqueuePolicy
+    , staticDequeuePolicy :: StaticDequeuePolicy
+    , staticFailurePolicy :: StaticFailurePolicy
+    }
+
+instance FromJSON StaticPolicies where
+    parseJSON = A.withObject "StaticPolicies" $ \obj -> do
+        staticEnqueuePolicy <- obj .: "enqueue"
+        staticDequeuePolicy <- obj .: "dequeue"
+        staticFailurePolicy <- obj .: "failure"
+        return StaticPolicies {..}
+
+data SendOrForward t = SendOrForward {
+      send    :: !t
+    , forward :: !t
+    }
+
+instance FromJSON t => FromJSON (SendOrForward t) where
+    parseJSON = A.withObject "SendOrForward" $ \obj -> do
+        send    <- obj .: "send"
+        forward <- obj .: "forward"
+        return SendOrForward {..}
+
+-- | An enqueue policy which can be described by JSON/YAML.
+newtype StaticEnqueuePolicy = StaticEnqueuePolicy {
+      getStaticEnqueuePolicy :: forall nid . MsgType nid -> [StaticEnqueue]
+    }
+
+instance FromJSON StaticEnqueuePolicy where
+    parseJSON = A.withObject "EnqueuePolicy" $ \obj -> do
+        announceBlockHeader <- obj .: "announceBlockHeader"
+        requestBlockHeaders <- obj .: "requestBlockHeaders"
+        requestBlocks       <- obj .: "requestBlocks"
+        transaction         <- obj .: "transaction"
+        mpc                 <- obj .: "mpc"
+        return $ StaticEnqueuePolicy $ \msg -> case msg of
```

Comment:
I like the translation to a function here, but then the `fromStaticPolicies` function is still somewhat complicated because you don't reuse the `Enqueue` type. I'm not sure why; if you did, you could simply unwrap that `StaticEnqueuePolicy` constructor in `fromStaticPolicies`, no?